### ISSUE TITLE
C4996: inet_ntop is deprecated [sfml-network]

### ISSUE THREAD

**Issue by PHEN-** [SFML/SFML#1023]
Using Visual Studio 14: MSVC 19.0.23506.0

Looks like I'm getting some deprecated warnings in sfml-network. Are these safe to ignore with _WINSOCK_DEPRECATED_NO_WARNINGS?

```
"C:\projects\gameengine\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\sfml-network.vcxproj" (default target) (16) ->
  C:\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\Ftp.cpp(621): warning C4244: '=': conversion from 'std::streamsize' to 'size_t', possible loss of data [C:\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\s
fml-network.vcxproj]
  C:\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\IpAddress.cpp(92): warning C4996: 'inet_ntoa': Use inet_ntop() or InetNtop() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to disable deprecated API warnings [C:\
projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\sfml-network.vcxproj]
  C:\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\IpAddress.cpp(184): warning C4996: 'inet_addr': Use inet_pton() or InetPton() instead or define _WINSOCK_DEPRECATED_NO_WARNINGS to disable deprecated API warnings [C:
\projects\gameengine\extinclude\SFML-Repo\src\SFML\Network\sfml-network.vcxproj]
```

**Comment by MarioLiebisch** [comment]
Yes, these are just a few of several deprecated warnings with MSVC 2015.

It's safe to ignore for now and shouldn't have any noticeable impact on anything you do with SFML.

I once started addressing them, but if I remember correctly, this is directly related to Windows XP compatibility (i.e. this might break building for Windows XP).

**Comment by PHEN-** [comment]
Windows XP has been end-of-life'd last year by Microsoft.  I'd be surprised if people are still asking for winxp support. 

Perhaps its good time to drop it?

**Comment by LaurentGomila** [comment]
Instead of blindly reporting what your compiler says, you should rather search why these functions are deprecated, and find if it's worth dropping support for an OS to do the replacement (ie. what's the benefit for SFML) ;)

**Comment by eXpl0it3r** [comment]
If you just want to ask whether a warning can be ignored or not, use the forum. The issue tracker should be for confirmed bugs only.

Any reason this hasn't been closed yet? @MarioLiebisch / @LaurentGomila

**Comment by binary1248** [comment]
```cpp
#include <SFML/Network/IpAddress.hpp>
#include <SFML/System/Thread.hpp>
#include <cstdio>

void entry(int i)
{
    char s[16];

    for (;;)
    {
        i = (i + 1) % 256;

        sf::IpAddress address(i, i, i, i);

        std::sprintf(s, "%u.%u.%u.%u", i, i, i, i);

        if (address.toString() != s)
            throw 1;
    }
}

int main()
{
    sf::Thread thread1(entry, 1);
    sf::Thread thread2(entry, 2);

    thread1.launch();
    thread2.launch();
}
```
This code causes an abort on FreeBSD and macOS. `inet_ntoa` is deprecated because in its original implementation it uses a statically allocated char buffer which means it is not thread-safe. Microsoft and glibc already worked around this by making the buffer thread local, but FreeBSD and macOS have not.

Safety-wise, there doesn't seem to be a problem with `inet_addr`. I guess it was deprecated along with `inet_ntoa` among other things because it doesn't support IPv6 addresses.

If anybody is wondering, writing our own versions of `inet_ntoa` and `inet_addr` wouldn't even be that complicated as long as we don't have to support those esoteric ways of specifying addresses. Try running `ping 0177.0x1` and you will see what I mean.

**Comment by JonnyPtn** [comment]
As we now set the define to suppress these warnings they no longer show up, so did we still want to do anything for this? I don't think SFML makes any guarantees about thread safety, so is it worth implementing our own versions of these functions still?
